controllers/front/webhook.php

```diff
@@ -251,20 +238,21 @@ public function processTransaction($transaction)
                         $orderStatusService->setOrderStatus($orderId, $paymentStatus);
 
                         $orderId = Order::getOrderByCartId((int)$apiPayment->metadata->cart_id);
-                    } elseif ($psPayment['method'] !== PaymentMethod::BANKTRANSFER
-                        && Tools::encrypt($cart->secure_key) === $apiPayment->metadata->secure_key
-                    ) {
+                    } elseif (Tools::encrypt($cart->secure_key) === $apiPayment->metadata->secure_key) {
                         $status = OrderStatusUtility::transformPaymentStatusToRefunded($apiPayment);
                         $paymentStatus = (int) Config::getStatuses()[$status];
 
-                        $transactionInfo = [
-                            'transactionId' => $transaction->id,
-                            'paymentMethod' => PaymentMethodUtility::getPaymentMethodName($transaction->method),
-                        ];
-
+                        if ($status === PaymentStatus::STATUS_PAID) {
+                            /** @var TransactionService $transactionService */
+                            $transactionService = $this->module->getContainer(TransactionService::class);
+                            $order = new Order($orderId);
+                            if (!$order->getOrderPayments()) {
+                                $transactionService->updateOrderTransaction($transaction->id, $order->reference);
+                            }
+                        }
```

Comment:
even your IDE should alert that it is duplicated code fragment from the one above - minimum effort you could do is just to create private function